Assignment 3.Rmd

```diff
@@ -0,0 +1,218 @@
+
+P6 <- P5 %>% group_by(cluster, week) %>%
+  summarise(meanCluster = mean(motivation))
+K6 <- 
+
+```
+
+Now it's time to do some visualization: 
+
+https://www.cs.uic.edu/~wilkinson/TheGrammarOfGraphics/GOG.html
+
+And you can see the range of available graphics in ggplot here:
+
+http://ggplot2.tidyverse.org/reference/index.html
+
+We are going to create a line plot similar to the one created in the school dropout paper we looked at in class (Bowers, 2010). It will have motivation on the Y-axis and weeks on the X-axis. To do this we will want our weeks variables to be treated as a number, but because it was created from a variable name it is currently being treated as a character variable. You can see this if you click on the arrow on the left of K6 in the Data pane. Week is designated by "chr". To convert it to numeric, we use the as.numeric command. 
+
+Likewise, since "cluster" is not numeric but rather a categorical label we want to convert it from an "integer" format to a "factor" format so that ggplot does not treat it as a number. We can do this with the as.factor() command.
+
+```{r}
+# change 'week' variable from chr to num - from factor to numeric
+P6$week <- as.numeric(P6$week)
+# change 'cluster' variable from int to fact - from numeric to factor
+P6$cluster <- as.factor(P6$cluster)
+
+P6$meanCluster <- as.numeric(P6$meanCluster)
+```
+
+Now we can plot our line plot using the ggplot command, "ggplot()".
+
+- The first argument in a ggplot is the dataframe we are using: K6
+- Next is what is called an aesthetic (aes), the aesthetic tells ggplot which variables to use and how to use them. Here we are using the variables "week" and "avg" on the x and y axes and we are going color these variables using the "cluster" variable
+- Then we are going to tell ggplot which type of plot we want to use by specifiying a "geom()", in this case a line plot: geom_line()
+- Finally we are going to clean up our axes labels: xlab("Week") & ylab("Average Motivation")
+
+```{r}
+# Use ggplot to plot ??  - the dataframe, the aesthetic - aes (x, week; y, meanCluster)
+# then color the cluster and choose geom_ to plot - line; changing axes labels (xlab, ylab)
+library(ggplot2)
+ggplot(P6, aes(x=week, y=meanCluster, color=cluster)) +
+         geom_line() +
+        xlab("Week") +
+        ylab("Average Motivation")
+
+```
+
+What patterns do you see in the plot?
+
+It would be useful to determine how many people are in each cluster. We can do this easily with dplyr.
+
+```{r}
+# For the 'lower cluster' students, week 2 and 4 should be examined to see why their avg. dipped to a clear low against the 'higher cluster' students clear high avg for same weeks
+?kmeans
+P7 <- kmeans(P6, 3)
+P7
+# Used kmeans() for summaries - How many in each cluster? cluster1 - 3, cluster2 - 3, cluster3 - 4
+```
```

Comment:
To get the number of students in each cluster, you should try to use the count function.